theano/tensor/nlinalg.py

```diff
@@ -726,3 +726,37 @@ def norm(x, ord):
             raise ValueError(0)
     elif ndim > 2:
         raise NotImplementedError("We don't support norm witn ndim > 2")
+
+
+class LogAbsDet(Op):
+    """Computes the logarithm of absolute determinant of a square
+    matrix M, log(abs(det(M))), on CPU. Avoids det(M) overflow/
+    underflow.
+
+    TODO: add GPU code!
+    """
+    def make_node(self, x):
+        x = theano.tensor.as_tensor_variable(x)
+        o = theano.tensor.scalar(dtype=x.dtype)
+        return Apply(self, [x], [o])
+
+    def perform(self, node, inputs, outputs):
+        try:
+            (x,) = inputs
+            (z,) = outputs
+            s = numpy.linalg.svd(x, compute_uv=False)
+            log_abs_det = numpy.sum(numpy.log(numpy.abs(s)))
```

Comment:
In fact, I think it is a good idea to keep this as an op. It will lower memory usage during the gradient on the CPU and will provide a gradient (the currrent svd op don't have a grad implemented).

but an easy way to have this on the GPU would be to make an optimization that will convert this op to a graph that reuse the elemwise on the GPU and make an svd op. For the gradient, MatrixInverse will also need a GPU op to have this work on the GPU. But the first step, is to get this working!